kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/wifi/HostapdManager.java

```diff
@@ -156,4 +154,58 @@ public static String getHostapdConfigFileName(String ifaceName) {
 		}
 		return sb.toString();
 	}
+	
+	private static String privGetHostapdConfigFileName(String ifaceName) {
+		StringBuilder sb = new StringBuilder();
+		if (OS_VERSION.equals(KuraConstants.Intel_Edison.getImageName() + "_" + KuraConstants.Intel_Edison.getImageVersion() + "_" + KuraConstants.Intel_Edison.getTargetName())) {
+			sb.append("/etc/hostapd/hostapd.conf");
+		} else {
+			sb.append("/tmp/hostapd-").append(ifaceName).append(".conf");
+		}
+		return sb.toString();
+	}
+	
+	private static File generateHostapdConfigFile(String ifaceName, Password passkey, WifiSecurity wifiSecurity) throws KuraException {
+		File retConfigFile = new File(privGetHostapdConfigFileName(ifaceName));
+		File configFile = new File(getHostapdConfigFileName(ifaceName));
+		if(!configFile.exists()) {
+			throw KuraException.internalError("Config file does not exist: " + configFile.getAbsolutePath());
+		}
+		BufferedReader br = null;
+		PrintWriter pw = null;	
+		try {
+			br = new BufferedReader(new FileReader(configFile));
```

Comment:
@ibinshtok be careful here: you opened a FileReader but not closed. I think this can lead to a leak. Same also for the following FileWriter